### PR TITLE
ATD-4758: Upgrade Base Image To Jdk17

### PR DESCRIPTION
**Pull request by Rapierian**
`This change includes the java version in the docker image tag name, and updates the documentation accordingly`

https://talamobile.atlassian.net/browse/ATD-4758

**Type of change**

Add an `x` to the ones that are appropriate.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

What tests did you run to verify your changes? Provide instructions so we can reproduce. Please also list any relevant
details for your test configuration

- [X] `Manually used the changes to publish JDK-17 based images to TalaEngineering`

##### Checklist:
Just add an `x` to the ones that are complete, so reviewers can determine if any remaining items are needed.

- [X] This review follows the [Tala Code Review Guideline](https://talamobile.atlassian.net/wiki/spaces/ATLAS/pages/1972818/Code+Review+and+Pull+Request+Process)
- [X] I have performed a self-review of my own code
- [X] I have made corresponding changes to the documentation. Updated documents are linked below:
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have updated the [External Libraries Tracking](https://goo.gl/a2iav1) spreadsheet with new libraries
that I added/updated/removed

### CODE REVIEW

**Review by LawrenceAsrikin — APPROVED** [review]
Nice! Let me run this locally to make sure everything still works in amd64 machine.

File: sbt-init.sh [hunk 1]
```diff
@@ -3,8 +3,8 @@
 source $SDKMAN_DIR/bin/sdkman-init.sh
 
 # Install Java and SBT - Reference: https://www.scala-sbt.org/1.x/docs/Installing-sbt-on-Linux.html
-# By default, it installs the latest version of Java 8 from OpenJDK (jdk.java.net)
-sdk install java 8.0.342-amzn
+# By default, it installs the latest version of Java 17 from OpenJDK (jdk.java.net)
+sdk install java 17.0.4-amzn
 sdk install sbt 1.7.1
```
Comment (LawrenceAsrikin) [inline sbt-init.sh:8]: We should update this file to load these values from the parameters instead of having them hardcoded.

**Review by LawrenceAsrikin — APPROVED** [review]
LG%C. One minor comment on sbt-init.

File: sbt-init.sh (hunk 1 above)
Comment (Rapierian) [reply sbt-init.sh:8]: Agreed. I tried to do initially but then realized that it was a larger change than I thought it would be.

**Pull request merged**